src/client_lib/pegasus_client_impl.h

```diff
@@ -247,13 +247,21 @@ class pegasus_client_impl : public pegasus_client
 
     static void init_error();
 
+    enum class async_scan_type : char
```

Comment:
Do not expose it to public in header, make it as a private of `class pegasus_scanner_impl` is enough?